pkg/daemon/daemon.go

```diff
@@ -42,42 +42,41 @@ type daemon struct {
 // It returns error in case of failure.
 func NewDaemon() (Daemon, error) {
 	glog.Info("daemon NewDaemon():")
-
 	daemonConfig := config.DaemonConfig{}
 	if err := daemonConfig.ReadConfig(); err != nil {
-		glog.Error(err)
 		return nil, err
 	}
 
 	if err := daemonConfig.ValidateConfig(); err != nil {
-		glog.Error(err)
 		return nil, err
 	}
 
 	podEventHandler := resEvenHandler.NewPodEventHandler()
-	client, err := k8sClient.NewK8sClient()
-
+	client, err := k8sClient.NewK8sClient(daemonConfig.Kube)
 	if err != nil {
-		glog.Error(err)
 		return nil, err
 	}
 
 	guidPool, err := guid.NewGuidPool(&daemonConfig.GuidPool, client)
 	if err != nil {
-		glog.Error(err)
 		return nil, err
 	}
 
-	if err = guidPool.InitPool(); err != nil {
-		glog.Error(err)
+	err = guidPool.InitPool()
+	if err != nil {
 		return nil, err
 	}
 
 	pluginLoader := sm.NewPluginLoader()
-	getSmClientFunc, err := pluginLoader.LoadPlugin(path.Join("/plugins", daemonConfig.SubnetManager.Plugin+".so"),
+
+	workDir, err := os.Getwd()
```

Comment:
nit: This, together with path.Join could go in a helper func which accepts subnetManagerConfig and returns the plugin path,
it would be cleaner here plus in the future it will be easier to add an "alternative plugin" base path